cumulus/client/consensus/aura/src/collators/lookahead.rs

```diff
@@ -315,13 +315,14 @@ where
 			let mut parent_header = initial_parent.header;
 			let overseer_handle = &mut params.overseer_handle;
 
-			// We mainly call this to inform users at genesis if there is a mismatch with the
-			// on-chain data.
-			collator.collator_service().check_block_status(parent_hash, &parent_header);
-
 			// This needs to change to support elastic scaling, but for continuously
 			// scheduled chains this ensures that the backlog will grow steadily.
 			for n_built in 0..2 {
+				// Do not try to build upon an unknown, pruned or bad block
+				if !collator.collator_service().check_block_status(parent_hash, &parent_header) {
+					break;
```

Comment:
Does this happen often ? I'd still add a trace here just in case.

Reply:
Not often at all, and `check_block_status()` has very detailed debug-level messages inside.